### PR TITLE
Add setting for Docker CLI executable path

### PR DESCRIPTION
**Pull request by karolz-ms**
Provides a workaround for [issue 2894](https://github.com/microsoft/vscode-docker/issues/2894)

### CODE REVIEW

File: src/commands/containers/attachShellContainer.ts [hunk 1]
```diff
@@ -46,7 +47,7 @@ export async function attachShellContainer(context: IActionContext, node?: Conta
             // If so use it, otherwise use sh
             try {
                 // If this succeeds, bash is present (exit code 0)
-                await execAsync(`docker exec -i ${node.containerId} sh -c "which bash"`);
+                await execAsync(`${dockerExePath()} exec -i ${node.containerId} sh -c "which bash"`);
```
Comment (philliphoff) [inline src/commands/containers/attachShellContainer.ts:50]: Should the Docker path be quoted to allow for paths with whitespace?

Comment (karolz-ms) [reply src/commands/containers/attachShellContainer.ts:50]: Good question. The user can include quotes in the path as necessary, for example the following works:

```json
"docker.dockerPath": "'/Users/karolz/docker local'"
```

(the above being hard link to `/usr/local/bin/docker`)

Comment (philliphoff) [reply src/commands/containers/attachShellContainer.ts:50]: Definitely true, but do we expect the user to *know* that they should do that?  (We don't generally require users to self-quote their paths, for example, in the task/launch JSON properties.)

Comment (karolz-ms) [reply src/commands/containers/attachShellContainer.ts]: Given that we do not expect it to be a common occurrence (that the path to Docker executable contains whitespace), I just updated the description of the setting to provide some guidance for the user.

File: package.nls.json [hunk 2]
```diff
@@ -188,6 +188,7 @@
     "vscode-docker.config.docker.showRemoteWorkspaceWarning": "Set to true to prompt to switch from \"UI\" extension mode to \"Workspace\" extension mode if an operation is not supported in UI mode.",
     "vscode-docker.config.docker.scaffolding.templatePath": "The path to use for scaffolding templates.",
     "vscode-docker.config.docker.showStartPage": "Show the Docker extension Start Page when a new update is released.",
+    "vscode-docker.config.docker.dockerPath": "Absolute path to Docker client executable ('docker' command). If the path contains whitespace, it need to be quoted appropriately.",
```
Comment (philliphoff) [inline package.nls.json:191]: Nit: "...it needs to be quoted..."

**Review by philliphoff — APPROVED** [review]

**Pull request merged**